auto_nag/scripts/vacant_triage_owner.py

```diff
@@ -0,0 +1,167 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+from jinja2 import Environment, FileSystemLoader
+from libmozdata.bugzilla import BugzillaProduct
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.nag_me import Nag
+from auto_nag.team_managers import TeamManagers
+from auto_nag.user_activity import UserActivity
+
+
+class TriageOwnerVacant(BzCleaner, Nag):
+    def __init__(self):
+        super(TriageOwnerVacant, self).__init__()
+        self.query_url = None
+
+    def description(self):
+        return "Components with triage owner need to be assigned"
+
+    def fetch_products(self):
+        data = []
+        include_fields = [
+            "name",
+            "is_active",
+            "components.id",
+            "components.name",
+            "components.team_name",
+            "components.triage_owner",
+            "components.is_active",
+        ]
+
+        def product_handler(product, data):
+            data.append(product)
+
+        BugzillaProduct(
+            product_names=self.get_products(),
+            include_fields=include_fields,
+            product_types=["accessible", "selectable", "enterable"],
+            product_handler=product_handler,
+            product_data=data,
+        ).wait()
+
+        return data
+
+    def nag_template(self):
+        return self.template()
+
+    def identify_vacant_components(self):
+        products = self.fetch_products()
+        triage_owners = set()
+        for product in products:
+            if not product["is_active"]:
+                continue
+            for component in product["components"]:
+                if not component["is_active"]:
+                    continue
+                triage_owners.add(component["triage_owner"])
+
+        inactive_users = UserActivity().check_users(triage_owners)
+        team_managers = TeamManagers()
+        vacant_components = []
+
+        for product in products:
+            if not product["is_active"]:
+                continue
+            for component in product["components"]:
+                if not component["is_active"]:
+                    continue
```

Comment:
You're doing the filtering about is_active twice (once to get list of triage owners, and once here), it'd be nice to do it just once at the beginning somehow

Reply:
We could have something like this at the beginning:
```python
products = [
    {
        **product,
        "components": [
            component
            for component in product["components"]
            if component["is_active"]
        ],
    }
    for product in self.fetch_products()
    if product["is_active"]
]
```
The issue is that we end up copying the lists, where the filtering is light and a few components usually will be excluded only (currently only 6). 

I guess in this case a little more comprehension is better than a little more optimization.